src/components/Weather.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useState } from 'react';
+
+const VITE_WEATHER_API_KEY = import.meta.env.VITE_WEATHER_API_KEY;
+
+function fetchWeather(
+  signal: AbortSignal,
+  setWeather: React.Dispatch<unknown>,
+) {
+  fetch(
+    `https://weerlive.nl/api/weerlive_api_v2.php?key=${VITE_WEATHER_API_KEY}&locatie=52.08718206955104,5.165697854286648`,
+    { signal },
+  )
+    .then((response) =>
+      response.json().then((data) => {
+        setWeather(data);
+      }),
+    )
+    .catch((error) => {
+      if (error.name === 'AbortError') {
+        setWeather(
+          JSON.parse('{"liveweer": [{ "temp": 0, "image": "bewolt"}]}'),
+        );
+      }
+    });
+}
```

Comment:
For API calls, ideally you create an API slice in the redux store. The benefit of that is that caching and error handling will be handled for you, and it will help with mocking if we want to do that in the future too. There are already some API slices you can take inspiration (read: copy code) from, and otherwise feel free to ask me if it's all a little vague.

Reply:
In `src/store/api/koala.ts`, we define types that the JSON data should be deserialized to. Should I define this for the weather API? I ask because it might be around 100 fields in multiple types and I feel that might be a bit much lol

Reply:
Oh yeah that is too much, just the fields you need.